### PR TITLE
Fix unexpected part name error when trying to drop a ignored detached part with zero copy replication

### LINKED ISSUE #48746: Broken detached parts cleanup does not work with zero-copy replication
```
Initialization failed, table will remain readonly. Error: Code: 233. DB::Exception: Unexpected part name: ignored_all_58400_58417_3 for format version: 1. (BAD_DATA_PART_NAME), Stack trace (when copying this message, always include the lines below):

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0xe0c66f5 in /usr/bin/clickhouse
1. ? @ 0x1451d6ac in /usr/bin/clickhouse
2. DB::MergeTreePartInfo::fromPartName(String const&, StrongTypedef<unsigned int, DB::MergeTreeDataFormatVersionTag>) @ 0x1451d183 in /usr/bin/clickhouse
3. DB::StorageReplicatedMergeTree::unlockSharedDataByID(String, String const&, String const&, String const&, String const&, std::shared_ptr<DB::ZooKeeperWithFaultInjection> const&, DB::MergeTreeSettings const&, Poco::Logger*, String const&, StrongTypedef<unsigned int, DB::MergeTreeDataFormatVersionTag>) @ 0x14018e28 in /usr/bin/clickhouse
4. DB::StorageReplicatedMergeTree::removeSharedDetachedPart(std::shared_ptr<DB::IDisk>, String const&, String const&, String const&, String const&, String const&, std::shared_ptr<DB::Context const> const&, std::shared_ptr<zkutil::ZooKeeper> const&) @ 0x1402f87f in /usr/bin/clickhouse
5. DB::StorageReplicatedMergeTree::removeDetachedPart(std::shared_ptr<DB::IDisk>, String const&, String const&) @ 0x1402f017 in /usr/bin/clickhouse
6. DB::MergeTreeData::clearOldBrokenPartsFromDetachedDirectory() @ 0x143c0d4d in /usr/bin/clickhouse
7. DB::ReplicatedMergeTreeAttachThread::runImpl() @ 0x146344fa in /usr/bin/clickhouse
8. DB::ReplicatedMergeTreeAttachThread::run() @ 0x14631b56 in /usr/bin/clickhouse
9. DB::BackgroundSchedulePoolTaskInfo::execute() @ 0x12586586 in /usr/bin/clickhouse
10. DB::BackgroundSchedulePool::threadFunction() @ 0x1258960a in /usr/bin/clickhouse
11. ? @ 0x1258a44e in /usr/bin/clickhouse
12. ThreadPoolImpl<std::thread>::worker(std::__list_iterator<std::thread, void*>) @ 0xe19a46a in /usr/bin/clickhouse
13. ? @ 0xe19fb21 in /usr/bin/clickhouse
14. ? @ 0x7fe7a20c4b43 in ?
15. ? @ 0x7fe7a2156a00 in ?
```

(this particular issue was worked around by https://github.com/ClickHouse/ClickHouse/pull/48730, but we still need to make the background cleanup work)

### PR DESCRIPTION
**Pull request by mlex**
I added the integration test [test_zero_copy_replication_drop_detached_part/test.py](https://github.com/ClickHouse/ClickHouse/pull/48862/files#diff-d2f05e57732e3f28be682af485315b7d2a2f59920ea3718f8112ba7fb352a2ec) which failed before this fix with the following exception message

```
E           Code: 233. DB::Exception: Received from 172.22.0.9:9000. DB::Exception: Unexpected part name: ignored_all_0_0_0 for format version: 1. Stack trace:
E
E           0. /home/ubuntu/clickhouse/contrib/llvm-project/libcxx/include/exception:134: std::exception::capture() @ 0x19bc7d02 in /usr/bin/clickhouse
E           1. /home/ubuntu/clickhouse/contrib/llvm-project/libcxx/include/exception:112: std::exception::exception[abi:v15000]() @ 0x19bc7ccd in /usr/bin/clickhouse
E           2. /home/ubuntu/clickhouse/base/poco/Foundation/src/Exception.cpp:27: Poco::Exception::Exception(String const&, int) @ 0x3194fae0 in /usr/bin/clickhouse
E           3. /home/ubuntu/clickhouse/src/Common/Exception.cpp:89: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x2268ae6e in /usr/bin/clickhouse
E           4. /home/ubuntu/clickhouse/src/Common/Exception.h:54: DB::Exception::Exception(String&&, int, bool) @ 0x19bbb00a in /usr/bin/clickhouse
E           5. /home/ubuntu/clickhouse/src/Common/Exception.h:81: DB::Exception::Exception<String const&, StrongTypedef<unsigned int, DB::MergeTreeDataFormatVersionTag>&>(int, FormatStringHelperImpl<std::type_identity<String const&>::type, std::type_identity<StrongTypedef<unsigned int, DB::MergeTreeDataFormatVersionTag>&>::type>, String const&, StrongTypedef<unsigned int, DB::MergeTreeDataFormatVersionTag>&) @ 0x2c4f318c in /usr/bin/clickhouse
E           6. /home/ubuntu/clickhouse/src/Storages/MergeTree/MergeTreePartInfo.cpp:23: DB::MergeTreePartInfo::fromPartName(String const&, StrongTypedef<unsigned int, DB::MergeTreeDataFormatVersionTag>) @ 0x2c4f0867 in /usr/bin/clickhouse
E           7. /home/ubuntu/clickhouse/src/Storages/StorageReplicatedMergeTree.cpp:8305: DB::(anonymous namespace)::getParentLockedBlobs(std::shared_ptr<DB::ZooKeeperWithFaultInjection> const&, String const&, String const&, StrongTypedef<unsigned int, DB::MergeTreeDataFormatVersionTag>, Poco::Logger*) @ 0x2bca90f7 in /usr/bin/clickhouse
```

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix unexpected part name error when trying to drop an ignored detached part with zero copy replication enabled. Fixes #48746

### CODE REVIEW

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -9035,6 +9039,11 @@ bool StorageReplicatedMergeTree::removeSharedDetachedPart(DiskPtr disk, const St
 
     NameSet files_not_to_remove;
 
+    auto part_info = DetachedPartInfo::parseDetachedPartName(disk, part_name, MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING);
+    if (!part_info.valid_name) {
```
Comment (alesapin) [inline src/Storages/StorageReplicatedMergeTree.cpp:9043]:
```suggestion
    if (!part_info.valid_name) 
    {
```

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 2]
```diff
@@ -9035,6 +9039,10 @@ bool StorageReplicatedMergeTree::removeSharedDetachedPart(DiskPtr disk, const St
 
     NameSet files_not_to_remove;
 
+    auto part_info = DetachedPartInfo::parseDetachedPartName(disk, part_name, MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING);
```
Comment (alesapin) [inline src/Storages/StorageReplicatedMergeTree.cpp:9042]: Maybe we need to add a comment why is it safe to always use `MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING`.

**Review by alesapin — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
Seems like we need to return back two lines deleted here? https://github.com/ClickHouse/ClickHouse/pull/48730

File: src/Storages/StorageReplicatedMergeTree.cpp (hunk 2 above)
Comment (mlex) [reply src/Storages/StorageReplicatedMergeTree.cpp:9042]: I'll add a comment. The same constant is already used in this [method a few lines](https://github.com/ClickHouse/ClickHouse/pull/48862/files#diff-0f4308161ab179a68f01cabfce4dc9e6e4e5f9c1e6c46c37d5adaeaad74e9913R9060) below, so we are not introducing any new assumptions here.

**Review by alesapin — APPROVED** [review]

**Pull request merged**